homeassistant/components/unifi/light.py

```diff
@@ -0,0 +1,196 @@
+"""Light platform for UniFi Network integration."""
+
+from __future__ import annotations
+
+from collections.abc import Awaitable, Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING, Any, cast
+
+from aiounifi.interfaces.api_handlers import APIHandler, ItemEvent
+from aiounifi.interfaces.devices import Devices
+from aiounifi.models.api import ApiItem
+from aiounifi.models.device import Device, DeviceSetLedStatus
+
+from homeassistant.components.light import (
+    ATTR_BRIGHTNESS,
+    ATTR_RGB_COLOR,
+    ColorMode,
+    LightEntity,
+    LightEntityDescription,
+    LightEntityFeature,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddConfigEntryEntitiesCallback
+
+from . import UnifiConfigEntry
+from .entity import (
+    UnifiEntity,
+    UnifiEntityDescription,
+    async_device_available_fn,
+    async_device_device_info_fn,
+)
+
+if TYPE_CHECKING:
+    from .hub import UnifiHub
+
+
+@callback
+def async_device_led_supported_fn(hub: UnifiHub, obj_id: str) -> bool:
+    """Check if device supports LED control."""
+    device = hub.api.devices[obj_id]
+    return device.supports_led_ring
+
+
+@callback
+def async_device_led_is_on_fn(hub: UnifiHub, device: Device) -> bool:
+    """Check if device LED is on."""
+    return device.led_override == "on"
+
+
+async def async_device_led_control_fn(
+    hub: UnifiHub, obj_id: str, **kwargs: Any
+) -> None:
+    """Control device LED."""
+    device = hub.api.devices[obj_id]
+
+    status = "on" if kwargs.get("turn_on", device.led_override == "on") else "off"
+
+    brightness = (
+        int((kwargs[ATTR_BRIGHTNESS] / 255) * 100)
+        if ATTR_BRIGHTNESS in kwargs
+        else device.led_override_color_brightness
+    )
+
+    color = (
+        f"#{kwargs[ATTR_RGB_COLOR][0]:02x}{kwargs[ATTR_RGB_COLOR][1]:02x}{kwargs[ATTR_RGB_COLOR][2]:02x}"
+        if ATTR_RGB_COLOR in kwargs
+        else device.led_override_color
+    )
+
+    await hub.api.request(
+        DeviceSetLedStatus.create(
+            device=device,
+            status=status,
+            brightness=brightness,
+            color=color,
+        )
+    )
+
+
+def _parse_hex_color(
+    hex_color: str | None, default: tuple[int, int, int]
+) -> tuple[int, int, int]:
+    """Parse hex color string to RGB tuple."""
+    if not hex_color:
+        return default
+
+    color_hex = hex_color.lstrip("#")
+    if len(color_hex) != 6:
+        return default
+
+    try:
+        rgb_values = [int(color_hex[i : i + 2], 16) for i in (0, 2, 4)]
+        return (rgb_values[0], rgb_values[1], rgb_values[2])
+    except ValueError:
+        return default
+
+
+@dataclass(frozen=True, kw_only=True)
+class UnifiLightEntityDescription[HandlerT: APIHandler, ApiItemT: ApiItem](
+    LightEntityDescription, UnifiEntityDescription[HandlerT, ApiItemT]
+):
+    """Class describing UniFi light entity."""
+
+    control_fn: Callable[..., Awaitable[None]]
```

Comment:
Should be something like `control_fn: Callable[[UnifiHub, str, bool], Coroutine[Any, Any, None]]`